maven/src/main/java/org/commonjava/maven/galley/maven/model/view/MavenPomView.java

```diff
@@ -1034,30 +1038,30 @@ protected boolean containsExpression( final String value )
      * @param activeProfileIds The profiles to consider active, whose elements should be treated as merged with
      *   the main POM body
      */
+
     public String resolveExpressions( final String value, final String... activeProfileIds )
     {
         if ( !containsExpression( value ) )
         {
-            //            logger.info( "No expressions in: '{}'", value );
             return value;
         }
 
-        synchronized ( this )
-        {
-            if ( ssi == null )
-            {
-                ssi = new StringSearchInterpolator();
-            }
-            ssi.addValueSource( new MavenPomViewVS( this, activeProfileIds ) );
-        }
+        List<String> prefixs = new ArrayList<>();
+        prefixs.add( PrefixAwareRecursionInterceptor.DEFAULT_START_TOKEN );
+        PrefixAwareRecursionInterceptor ri = new PrefixAwareRecursionInterceptor( prefixs );
+        return resolveExpressions( value, ri, activeProfileIds );
+    }
 
+    public String resolveExpressions( final String value, final RecursionInterceptor ri,
+                                      final String... activeProfileIds )
+    {
+        StringSearchInterpolator ssi = new StringSearchInterpolator();
+        ssi.addValueSource( new MavenPomViewVS( this, ri, activeProfileIds ) );
         try
         {
-            String result = ssi.interpolate( value );
-            //            logger.info( "Resolved '{}' to '{}'", value, result );
+            String result = ssi.interpolate( value, ri );
```

Comment:
I think you need a null check on `ri` here, and construct a RecursionInterceptor if it is null.